data/senml_cbor.c

```diff
@@ -139,6 +139,8 @@ static int prv_parseItem(const uint8_t *buffer, size_t bufferLen, senml_record_t
             bvSeen = true;
             memcpy(baseValue, &data, sizeof(lwm2m_data_t));
             /* Convert explicit 0 to implicit 0 */
+            _Pragma("GCC diagnostic push");
+            _Pragma("GCC diagnostic ignored \"-Wfloat-equal\"");
```

Comment:
I really don't like the compiler specific _Pragma. Not everyone uses gcc.

Reply:
I think there are just 3 equality checks in there outside of the examples and tests. I'm more concerned with the base library than those.

1. Check for NaN which could be replaced by isnan(), I was just targeting C90 when I wrote it.
2. Check whether or not anything would be lost converting double to float (and back). If not, then it can be encoded single precision. Another method would probably need to break it down into the exponent and mantissa and check those. That may need to make assumptions about the underlying hardware representation, although that assumption is probably already being made.
3. Explicitly check for 0. Something like fabsf(a - b) < FLT_EPSILON would involve extra floating point computations and classify anything less than FLT_EPSILON as 0, which would be incorrect. Again, checks on the mantissa and exponent could be done, but be careful that -0 and +0 are treated the same. That would also likely be more computation than a simple compare.

Reply:
I think we could use `fpclassify` since we rely on C11.

Reply:
> I think we could use fpclassify since we rely on C11.

Will do this in a separate PR. There are already some Pragma in the code to mute floating point comparison compiler warnings. They can be removed as well.